engine/classes/Elgg/ViewsService.php

```diff
@@ -561,8 +589,11 @@ public function unextendView($view, $view_extension) {
 		if (!isset($this->extensions[$view])) {
 			return false;
 		}
+		
+		$extensions = $this->extensions[$view];
+		unset($extensions[self::BASE_VIEW_PRIORITY]); // we do not want the base view to be removed from the list
```

Comment:
What does this prevent? I guess I'm not getting it.

Reply:
see test case https://github.com/Elgg/Elgg/pull/11231/files#diff-62ab3d7df7906064c85f6ac437fe6128R231

currently you can remove the base view from its own extension list
just do
```
elgg_extend_view('a', 'b');
elgg_unextend_view('a', 'a');
elgg_view('a'); // output will be only b
```